### ISSUE TITLE
[2.7.0] SQL error when trying to edit entry

### ISSUE THREAD

**Issue by Mosnar** [spicywebau/craft-neo#315]
#### Description
It looks like the latest update violates the GROUP BY aggregation rules in some MySQL installations. I receive the following error when trying to view an edit entry page:

```
PDOException: SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'staging_mysite.elements.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by in /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php:1290
Stack trace:
#0 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(1290): PDOStatement->execute()
#1 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(1158): yii\db\Command->internalExecute('SELECT `element...')
#2 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(399): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /home/forge/mysite./vendor/yiisoft/yii2/db/Query.php(237): yii\db\Command->queryAll()
#4 /home/forge/mysite./vendor/craftcms/cms/src/db/Query.php(152): yii\db\Query->all(NULL)
#5 /home/forge/mysite./vendor/craftcms/cms/src/elements/db/ElementQuery.php(1471): craft\db\Query->all(NULL)
#6 /home/forge/mysite./vendor/spicyweb/craft-neo/src/elements/db/BlockQuery.php(259): craft\elements\db\ElementQuery->all(NULL)
#7 /home/forge/mysite./vendor/spicyweb/craft-neo/src/Field.php(903): benf\neo\elements\db\BlockQuery->all()
#8 /home/forge/mysite./vendor/spicyweb/craft-neo/src/Field.php(348): benf\neo\Field->_getInputHtml(Object(benf\neo\elements\db\BlockQuery), Object(craft\elements\Entry))
#9 /home/forge/mysite./vendor/twig/twig/src/Extension/CoreExtension.php(1499): benf\neo\Field->getInputHtml(Object(benf\neo\elements\db\BlockQuery), Object(craft\elements\Entry))
#10 /home/forge/mysite./vendor/craftcms/cms/src/helpers/Template.php(96): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(benf\neo\Field), 'getInputHtml', Array, 'method', false, false)
#11 /home/forge/mysite./storage/runtime/compiled_templates/ee/eee50fe7005fe48617189460ebd7b2013a3e487c44e74675c33aa3bf5df5d5ad.php(87): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(benf\neo\Field), 'getInputHtml', Array, 'method')
#12 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_6213374efb3887fbe1c9c4329651eebd5d82d0325b724660b46ff1c0e75e9837->doDisplay(Array, Array)
#13 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#14 /home/forge/mysite./storage/runtime/compiled_templates/da/da98594bb9e855b7d2a139b4d18a2b42eaa9570fff51223f341fb3e177ec45a6.php(59): Twig\Template->display(Array)
#15 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_4fbcbe1497d391a3d4b575ac55135c82b6263cea9cc424806140c0472f04cf6e->doDisplay(Array, Array)
#16 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#17 /home/forge/mysite./storage/runtime/compiled_templates/fc/fcca93952c854aa688cf6ed5a77d0b3b3712e95392422a18bcc2105fbec1fbbf.php(79): Twig\Template->display(Array)
#18 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_f44be5ca8a6315ba2a8f9d12e26d05df727493aa4bfeb572ba3bdc9b48f8ee41->doDisplay(Array, Array)
#19 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#20 /home/forge/mysite./storage/runtime/compiled_templates/5b/5b6c3061a2c5e6d8e25aaca292b2912c86592468602e47b4e66eeaceab9ba893.php(156): Twig\Template->display(Array)
#21 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_176629b1edabc1148738f46fc3fa037c8fcb6e83125e958c60b32e16bc6f5c94->block_content(Array, Array)
#22 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(411): Twig\Template->displayBlock('content', Array, Array)
#23 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->block_main(Array, Array)
#24 /home/forge/mysite./vendor/twig/twig/src/Template.php(147): Twig\Template->displayBlock('main', Array, Array, false)
#25 /home/forge/mysite./storage/runtime/compiled_templates/c4/c4709e51ab503e0c901bea607e5f3a1623a816620a4bfcbd64a23cadb7ad0deb.php(548): Twig\Template->displayParentBlock('main', Array, Array)
#26 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_757dad7dae38352db354cc71d469d5bc4e1c0e055afec86c13e03ed71aa50dbb->block_main(Array, Array)
#27 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(266): Twig\Template->displayBlock('main', Array, Array)
#28 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->block_body(Array, Array)
#29 /home/forge/mysite./storage/runtime/compiled_templates/b7/b76cb2af25308bc0d477f487602539a8d2b4f8b7e1e2dc80d448ef943228589d.php(75): Twig\Template->displayBlock('body', Array, Array)
#30 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_3b4a24d2eee5c5759a9efb6e1d207d267639a30080113c59b8d733352eb7a463->doDisplay(Array, Array)
#31 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#32 /home/forge/mysite./storage/runtime/compiled_templates/da/dacbfb0681cf2345ae7e449c996ce930054c57441b4b1d8041d955845e2dfc2e.php(56): Twig\Template->display(Array, Array)
#33 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_483620c2220a40418897bdd7808a2ad6109341c039e528afbbc65b61981f1dea->doDisplay(Array, Array)
#34 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#35 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(138): Twig\Template->display(Array, Array)
#36 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->doDisplay(Array, Array)
#37 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#38 /home/forge/mysite./storage/runtime/compiled_templates/c4/c4709e51ab503e0c901bea607e5f3a1623a816620a4bfcbd64a23cadb7ad0deb.php(222): Twig\Template->display(Array, Array)
#39 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_757dad7dae38352db354cc71d469d5bc4e1c0e055afec86c13e03ed71aa50dbb->doDisplay(Array, Array)
#40 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#41 /home/forge/mysite./storage/runtime/compiled_templates/5b/5b6c3061a2c5e6d8e25aaca292b2912c86592468602e47b4e66eeaceab9ba893.php(120): Twig\Template->display(Array, Array)
#42 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_176629b1edabc1148738f46fc3fa037c8fcb6e83125e958c60b32e16bc6f5c94->doDisplay(Array, Array)
#43 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#44 /home/forge/mysite./vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#45 /home/forge/mysite./vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#46 /home/forge/mysite./vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#47 /home/forge/mysite./vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('entries/_edit', Array)
#48 /home/forge/mysite./vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('entries/_edit', Array)
#49 /home/forge/mysite./vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('entries/_edit', Array, 'cp')
#50 /home/forge/mysite./vendor/craftcms/cms/src/controllers/EntriesController.php(248): craft\web\Controller->renderTemplate('entries/_edit', Array)
#51 [internal function]: craft\controllers\EntriesController->actionEditEntry(Object(craft\models\Section), 895, NULL, NULL, Object(craft\models\Site), Object(craft\elements\Entry))
#52 /home/forge/mysite./vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#53 /home/forge/mysite./vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#54 /home/forge/mysite./vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('edit-entry', Array)
#55 /home/forge/mysite./vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('edit-entry', Array)
#56 /home/forge/mysite./vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('entries/edit-en...', Array)
#57 /home/forge/mysite./vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('entries/edit-en...', Array)
#58 /home/forge/mysite./vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#59 /home/forge/mysite./vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#60 /home/forge/mysite./web/index.php(17): yii\base\Application->run()
#61 {main}

Next yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'staging_mysite.elements.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
The SQL being executed was: SELECT `elements`.`id`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `neoblocks`.`fieldId`, `neoblocks`.`ownerId`, `neoblocks`.`typeId`, `neoblocks`.`sortOrder`, `content`.`id` AS `contentId`, 
<snipped my fields>
`structureelements`.`root`, `structureelements`.`lft`, `structureelements`.`rgt`, `structureelements`.`level`, `structureelements`.`structureId`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`, `structureelements`.`structureId`
FROM `craft_elements` `elements`
INNER JOIN `craft_neoblocks` `neoblocks` ON `neoblocks`.`id` = `elements`.`id`
INNER JOIN `craft_elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `craft_content` `content` ON `content`.`elementId` = `elements`.`id`
LEFT JOIN `craft_structureelements` `structureelements` ON (`structureelements`.`elementId` = `elements`.`id`) AND (EXISTS (SELECT *
FROM `craft_structures`
WHERE (`id` = `structureelements`.`structureId`) AND (`dateDeleted` IS NULL)))
WHERE (`neoblocks`.`fieldId`='41') AND (`neoblocks`.`ownerId`='895') AND (`elements`.`archived`=FALSE) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)) `subquery`
INNER JOIN `craft_neoblocks` `neoblocks` ON `neoblocks`.`id` = `subquery`.`elementsId`
INNER JOIN `craft_elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `craft_elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `craft_content` `content` ON `content`.`id` = `subquery`.`contentId`
LEFT JOIN `craft_structureelements` `structureelements` ON (`structureelements`.`elementId` = `subquery`.`elementsId`) AND (`structureelements`.`structureId` = `subquery`.`structureId`)
GROUP BY `neoblocks`.`sortOrder`
ORDER BY `neoblocks`.`sortOrder` in /home/forge/mysite./vendor/yiisoft/yii2/db/Schema.php:674
Stack trace:
#0 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(1295): yii\db\Schema->convertException(Object(PDOException), 'SELECT `element...')
#1 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(1158): yii\db\Command->internalExecute('SELECT `element...')
#2 /home/forge/mysite./vendor/yiisoft/yii2/db/Command.php(399): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /home/forge/mysite./vendor/yiisoft/yii2/db/Query.php(237): yii\db\Command->queryAll()
#4 /home/forge/mysite./vendor/craftcms/cms/src/db/Query.php(152): yii\db\Query->all(NULL)
#5 /home/forge/mysite./vendor/craftcms/cms/src/elements/db/ElementQuery.php(1471): craft\db\Query->all(NULL)
#6 /home/forge/mysite./vendor/spicyweb/craft-neo/src/elements/db/BlockQuery.php(259): craft\elements\db\ElementQuery->all(NULL)
#7 /home/forge/mysite./vendor/spicyweb/craft-neo/src/Field.php(903): benf\neo\elements\db\BlockQuery->all()
#8 /home/forge/mysite./vendor/spicyweb/craft-neo/src/Field.php(348): benf\neo\Field->_getInputHtml(Object(benf\neo\elements\db\BlockQuery), Object(craft\elements\Entry))
#9 /home/forge/mysite./vendor/twig/twig/src/Extension/CoreExtension.php(1499): benf\neo\Field->getInputHtml(Object(benf\neo\elements\db\BlockQuery), Object(craft\elements\Entry))
#10 /home/forge/mysite./vendor/craftcms/cms/src/helpers/Template.php(96): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(benf\neo\Field), 'getInputHtml', Array, 'method', false, false)
#11 /home/forge/mysite./storage/runtime/compiled_templates/ee/eee50fe7005fe48617189460ebd7b2013a3e487c44e74675c33aa3bf5df5d5ad.php(87): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(benf\neo\Field), 'getInputHtml', Array, 'method')
#12 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_6213374efb3887fbe1c9c4329651eebd5d82d0325b724660b46ff1c0e75e9837->doDisplay(Array, Array)
#13 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#14 /home/forge/mysite./storage/runtime/compiled_templates/da/da98594bb9e855b7d2a139b4d18a2b42eaa9570fff51223f341fb3e177ec45a6.php(59): Twig\Template->display(Array)
#15 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_4fbcbe1497d391a3d4b575ac55135c82b6263cea9cc424806140c0472f04cf6e->doDisplay(Array, Array)
#16 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#17 /home/forge/mysite./storage/runtime/compiled_templates/fc/fcca93952c854aa688cf6ed5a77d0b3b3712e95392422a18bcc2105fbec1fbbf.php(79): Twig\Template->display(Array)
#18 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_f44be5ca8a6315ba2a8f9d12e26d05df727493aa4bfeb572ba3bdc9b48f8ee41->doDisplay(Array, Array)
#19 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#20 /home/forge/mysite./storage/runtime/compiled_templates/5b/5b6c3061a2c5e6d8e25aaca292b2912c86592468602e47b4e66eeaceab9ba893.php(156): Twig\Template->display(Array)
#21 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_176629b1edabc1148738f46fc3fa037c8fcb6e83125e958c60b32e16bc6f5c94->block_content(Array, Array)
#22 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(411): Twig\Template->displayBlock('content', Array, Array)
#23 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->block_main(Array, Array)
#24 /home/forge/mysite./vendor/twig/twig/src/Template.php(147): Twig\Template->displayBlock('main', Array, Array, false)
#25 /home/forge/mysite./storage/runtime/compiled_templates/c4/c4709e51ab503e0c901bea607e5f3a1623a816620a4bfcbd64a23cadb7ad0deb.php(548): Twig\Template->displayParentBlock('main', Array, Array)
#26 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_757dad7dae38352db354cc71d469d5bc4e1c0e055afec86c13e03ed71aa50dbb->block_main(Array, Array)
#27 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(266): Twig\Template->displayBlock('main', Array, Array)
#28 /home/forge/mysite./vendor/twig/twig/src/Template.php(184): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->block_body(Array, Array)
#29 /home/forge/mysite./storage/runtime/compiled_templates/b7/b76cb2af25308bc0d477f487602539a8d2b4f8b7e1e2dc80d448ef943228589d.php(75): Twig\Template->displayBlock('body', Array, Array)
#30 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_3b4a24d2eee5c5759a9efb6e1d207d267639a30080113c59b8d733352eb7a463->doDisplay(Array, Array)
#31 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#32 /home/forge/mysite./storage/runtime/compiled_templates/da/dacbfb0681cf2345ae7e449c996ce930054c57441b4b1d8041d955845e2dfc2e.php(56): Twig\Template->display(Array, Array)
#33 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_483620c2220a40418897bdd7808a2ad6109341c039e528afbbc65b61981f1dea->doDisplay(Array, Array)
#34 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#35 /home/forge/mysite./storage/runtime/compiled_templates/d6/d6e97d1eaa450691bcc3a10fe38656cba475af30b33507df955d030478dd0f0d.php(138): Twig\Template->display(Array, Array)
#36 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_eae2819ec2f89493772e2215839e8b1410eb91650373154d4d50bdd3e1cd4ecf->doDisplay(Array, Array)
#37 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#38 /home/forge/mysite./storage/runtime/compiled_templates/c4/c4709e51ab503e0c901bea607e5f3a1623a816620a4bfcbd64a23cadb7ad0deb.php(222): Twig\Template->display(Array, Array)
#39 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_757dad7dae38352db354cc71d469d5bc4e1c0e055afec86c13e03ed71aa50dbb->doDisplay(Array, Array)
#40 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#41 /home/forge/mysite./storage/runtime/compiled_templates/5b/5b6c3061a2c5e6d8e25aaca292b2912c86592468602e47b4e66eeaceab9ba893.php(120): Twig\Template->display(Array, Array)
#42 /home/forge/mysite./vendor/twig/twig/src/Template.php(407): __TwigTemplate_176629b1edabc1148738f46fc3fa037c8fcb6e83125e958c60b32e16bc6f5c94->doDisplay(Array, Array)
#43 /home/forge/mysite./vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#44 /home/forge/mysite./vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#45 /home/forge/mysite./vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#46 /home/forge/mysite./vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#47 /home/forge/mysite./vendor/craftcms/cms/src/web/View.php(397): Twig\Environment->render('entries/_edit', Array)
#48 /home/forge/mysite./vendor/craftcms/cms/src/web/View.php(458): craft\web\View->renderTemplate('entries/_edit', Array)
#49 /home/forge/mysite./vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('entries/_edit', Array, 'cp')
#50 /home/forge/mysite./vendor/craftcms/cms/src/controllers/EntriesController.php(248): craft\web\Controller->renderTemplate('entries/_edit', Array)
#51 [internal function]: craft\controllers\EntriesController->actionEditEntry(Object(craft\models\Section), 895, NULL, NULL, Object(craft\models\Site), Object(craft\elements\Entry))
#52 /home/forge/mysite./vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#53 /home/forge/mysite./vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#54 /home/forge/mysite./vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('edit-entry', Array)
#55 /home/forge/mysite./vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('edit-entry', Array)
#56 /home/forge/mysite./vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('entries/edit-en...', Array)
#57 /home/forge/mysite./vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('entries/edit-en...', Array)
#58 /home/forge/mysite./vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#59 /home/forge/mysite./vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#60 /home/forge/mysite./web/index.php(17): yii\base\Application->run()
#61 {main}
Additional Information:
Array
(
    [0] => 42000
    [1] => 1055
    [2] => Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'staging_mysite.elements.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
)
```

Presumably I could fix this by updating the sql_mode like we used to have to do with Craft 2; however, it's probably better to fix this in the plugin.

#### Other information
- Neo version: 2.7.0
- Craft version: 3.4.10.1
- MySQL version: 5.7.29

**Comment by pvldigital** [comment]
Hi @Mosnar,

Ah, the group by clause was added for postgres since it was throwing an error. I'll make a change so it's only added for postgres.

**Comment by wrux** [comment]
Hi @pvldigital will there be a patch release for this?

**Comment by pvldigital** [comment]
@wrux Yes, I'm just testing it now. Should be released within the next hour.

**Comment by wrux** [comment]
@pvldigital Nice, I was getting the same error in my test site and your change seems to fix the issue

**Comment by pvldigital** [comment]
@wrux @Mosnar fix has been made for this issue in v2.7.1.

**Comment by Mosnar** [comment]
Sure enough! Thanks for the quick support.

**Comment by Mosnar** [comment]
@pvldigital Erk! Problem 2: All of my blocks are showing twice, both in the CP and on the frontend. Let me know if I can provide any additional info to help.

**Comment by pvldigital** [comment]
@Mosnar hmm that's strange. what happens when you remove the extra blocks and resave?

**Comment by pvldigital** [comment]
Actually i do remember something similar happen, where some blocks appeared out of nowhere after the migrations. I found that they we remnant/duplicated/revision blocks that wasn't completely removed from the DB after content changes was made. This was mainly because of how the structure was created previously which has been modified.

Removing them and then resaving should fix the issue. If not, can you send the DB and composer files to plugins@spicyweb.com.au and i'll have a look at it.

**Comment by Mosnar** [comment]
It's a little complicated to explain, so I'll try to illustrate it. These are my blocks:

```
A1
A2 (duplicate of A1)
B1
B2
C1
C2
D1
D2
```

If I delete A1, the backend shows the same thing minus the single block I deleted; however, the frontend now only shows A2 and does not render the other blocks.

If I delete A2, the backend corrects itself and all duplicates are removed; however, due to the delta-saving, only A1 appears on the frontend. If I manually edit each block, the content reappears.

**Comment by pvldigital** [comment]
hmm that is strange. I'll see if I can replicate the issue.

**Comment by pvldigital** [comment]
@Mosnar are you using eagerloading at all?

**Comment by Mosnar** [comment]
@pvldigital Nope

**Comment by pvldigital** [comment]
@Mosnar I was unable to replicate the issue but I've added some changes to the latest commit to check and rebuild if there has been any block deletions.

Is it possible for you to send your DB and composer files? So I can test and see what the issue might be.

**Comment by Mosnar** [comment]
@pvldigital Unfortunately I can't share the database at this point due to client privacy. I did find that your latest update fixed the block duplication issue; however, now my blocks are out of order both on the frontend and backend. Resaving doesn't help

**Comment by pvldigital** [comment]
Ah no worries. In that case can you check the `neoblocks` table within the DB? Can you see if the `sortOrder` columns is `null`?

**Comment by Mosnar** [comment]
@pvldigital I just did a fresh re-run of the update and found that the neoblocks that were appearing out of order did not have a sortOrder; however, there are a handful of blocks that do have a sort order set.

**Comment by Mosnar** [comment]
I believe I found the problem with the migration:
https://github.com/spicywebau/craft-neo/blob/master/src/migrations/m200313_015120_structure_update.php#L41

1) `'structureelements.id as sId'` should be `'structureelements.elementId as sId'` on line 39
2) structureelements is being joined on `id` rather than `elementId` on line 41

I updated the migration as such:
```php

        $query = (new Query())
            ->select(['elements.id', 'elements.type', 'structureelements.elementId as sId', 'structureelements.lft'])
            ->from('{{%elements}} elements')
            ->leftJoin('{{%structureelements}} structureelements', '[[elements.id]] = [[structureelements.elementId]]')
            ->where(['elements.type' => Block::class])
            ->andWhere('structureelements.lft IS NOT NULL')
            ->limit(null);

        $elements = $query->all($this->db);

        foreach ($elements as $el) {
            $this->update('{{%neoblocks}}', ['sortOrder' => (int)$el['lft']],
                ['id' => (int)$el['sId']]);
        }
```

and re-ran it, and found all of the sort orders match their original `lft` values.

**Comment by pvldigital** [comment]
ahh thanks for looking into that. I knew it was something to do with the migration but didn't notice that it should've been the elementId instead of id. Thanks for that!

**Comment by pvldigital** [comment]
@Mosnar your fix has been applied and release in v2.7.2.1. thanks again for looking into that.
